src/components/Markings/ImageWithSvgOverlay.js

```diff
@@ -10,47 +11,61 @@ import {
     Rect
 } from 'react-native-svg'
 import R from 'ramda'
-import { connect } from 'react-redux'
+import { BlurView } from 'react-native-blur';
+import Icon from 'react-native-vector-icons/FontAwesome'
 import SubjectLoadingIndicator from '../common/SubjectLoadingIndicator';
 
-const mapStateToProps = (state) => {
-    const subjectDimensions = state.classifier.subjectDimensions[state.classifier.subject.id]
-    return {
-        subjectDimensions: subjectDimensions ? subjectDimensions : {naturalHeight: 1, naturalWidth: 1}
-    }
-}
-
 class ImageWithSvgOverlay extends Component {
 
     constructor(props) {
         super(props)
-        this.count = 0
+
         this.state = {
-            imageIsLoaded: false,
-            imageNativeWidth: 1,
-            imageNativeHeight: 1,
-            clientWidth: 1,
-            clientHeight: 1,
+            scale: new Animated.Value(1),
+            containerDimensions: {
+                width: 1,
+                height: 1
+            },
         }
 
         this.onImageLayout = this.onImageLayout.bind(this)
+        this.animateScale = this.animateScale.bind(this)
+    }
+
+    componentDidUpdate(prevProps, prevState) {
+        if (prevProps.imageIsLoaded !== this.props.imageIsLoaded && !this.props.imageIsLoaded) {
+            this.setState({scale: new Animated.Value(0.8)})
+        }
+        if (!R.equals(prevProps.subjectDimensions, this.props.subjectDimensions) || !R.equals(prevState.containerDimensions, this.state.containerDimensions)) {
+            const { naturalHeight, naturalWidth } = this.props.subjectDimensions
+            const { height: containerHeight, width: containerWidth } = this.state.containerDimensions
```

Comment:
So the object `this.state.containerDimensions` has two props: `height` and `width` but because those names are generic, this syntax lets you assign those values to `containerHeight` and `containerWidth` respectively.